input/witnessgen.go

```diff
@@ -47,116 +47,118 @@ type StandardWitnessType uint16
 // WitnessType interface.
 var _ WitnessType = (StandardWitnessType)(0)
 
+// NOTE: When adding a new `StandardWitnessType`, also update the `WitnessType`
+// protobuf enum and the `allWitnessTypes` map in the `walletrpc` package.
 const (
 	// CommitmentTimeLock is a witness that allows us to spend our output
 	// on our local commitment transaction after a relative lock-time
 	// lockout.
-	CommitmentTimeLock StandardWitnessType = 0
+	CommitmentTimeLock StandardWitnessType = 1
```

Comment:
We can't just change the value of those constants! They are used for storing things in databases (watchtower, breach arbiter), so they can't just have a new meaning suddenly. We should instead explain in the mapping function why they are misaligned (which is why we need the mapping function in the first place, otherwise we could just do `rpcType = witnessType +1`).

Reply:
Yikes! Thanks for catching this. I had considered the backward compatibility of the protobuf definition, but not the databases. I have reverted the renumbering, and added further commentary above the `allWitnessTypes` map explaining why this mapping is necessary.

One solution would be to renumber the enum *and* run migrations in the relevant databases. But it doesn't seem worth the effort just to have the numbering match up between the enums. What do you think?